processor/src/test/java/org/mapstruct/ap/test/value/DefaultOrderMapper.java

```diff
@@ -0,0 +1,39 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.value;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.ValueMapping;
+import org.mapstruct.ValueMappingType;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Sjaak Derksen
+ */
+@Mapper
+public interface DefaultOrderMapper {
+
+    DefaultOrderMapper INSTANCE = Mappers.getMapper( DefaultOrderMapper.class );
+
+    OrderDto orderEntityToDto(OrderEntity order);
+
+
+    @ValueMapping( valueMappingType = ValueMappingType.DEFAULT_AFTER_APPLYING_NAME_BASED_MAPPINGS, target = "DEFAULT" )
```

Comment:
So is this meant as a "catch all unmatched" mapping for all unmapped ones? If so, how about this, using the more explicit `@EnumMapping`:

    @EnumMapping( source = "*", target = "DEFAULT" )

Much easier to read, isn't it? That's the advantage of the explicit `@EnumMapping`: We can use special values such as "null" and "*" because these are no valid enum constants, hence there are no ambiguities.